app/knight/knight.py

```diff
@@ -0,0 +1,51 @@
+from __future__ import annotations
+
+
+class Knight:
+    def __init__(
+            self,
+            knight_dict: dict
+    ) -> None:
+        self.name = knight_dict.get("name")
+        self.power = knight_dict.get("power")
+        self.hp = knight_dict.get("hp")
+        self.weapon = knight_dict.get("weapon")
+        self.armour = knight_dict.get("armour")
+        self.potion = knight_dict.get("potion")

```

Comment:
Consider adding a separate module where you will place different classes for `Weapon`, `Armour`, and `Potion`. It will make the code more structured and will add an opportunity to look at the initialization of each class.